### PR TITLE
Disallow insecure randomness functions

### PR DESCRIPTION
**Pull request by compwright**

### CODE REVIEW

File: disallowed-insecure-calls.neon [hunk 1]
```diff
@@ -67,3 +67,21 @@ parameters:
 		-
 			function: 'mysqli::real_query()'
 			message: 'use PDO::prepare() with variable binding/parametrized queries to prevent SQL Injection vulnerability'
+		-
+			function: 'rand()'
+			message: 'not cryptographically secure, use random_int() instead'
+		-
+			function: 'srand()'
+			message: 'not cryptographically secure, use random_int() instead'
```
Comment (spaze) [inline disallowed-insecure-calls.neon:75]: Although I see calling this function as a red flag, I'd leave this out, the function itself doesn't really do anything insecure.

File: disallowed-insecure-calls.neon [hunk 2]
```diff
@@ -67,3 +67,21 @@ parameters:
...
+		-
+			function: 'mt_rand()'
+			message: 'not cryptographically secure, use random_int() instead'
+		-
+			function: 'mt_srand()'
+			message: 'not cryptographically secure, use random_int() instead'
```
Comment (spaze) [inline disallowed-insecure-calls.neon:81]: Same here

**Review by spaze — CHANGES_REQUESTED** [review]
Thanks, this is a good idea! I'd only leave out the seeding functions, and please move the functions in the `disallowedFunctionCalls` section, you have it in methods.

File: disallowed-insecure-calls.neon (hunk 1 above)
Comment (spaze) [reply disallowed-insecure-calls.neon]: Removed

File: disallowed-insecure-calls.neon (hunk 2 above)
Comment (spaze) [reply disallowed-insecure-calls.neon]: Removed

**Review by spaze — APPROVED** [review]

**Comment by spaze** [comment]
Thank you! I've made some small changes:
- removed the seeding functions
- moved the new block of functions to the `disallowedFunctionCalls` section
- slightly changed the message
- added tests

and pushed the Merge button

**Comment by compwright** [comment]
Thank you!

**Comment by compwright** [comment]
Will there be a release soon?

**Comment by spaze** [comment]
> Will there be a release soon?

Hopefully later today :-) I want to sneak in one more thing I'm working on now.

**Comment by spaze** [comment]
Just released the updated config in [2.11.0](https://github.com/spaze/phpstan-disallowed-calls/releases/tag/v2.11.0), let me know if that's ok. Thanks for the contribution 👍

**Pull request merged**